### ISSUE TITLE
Source key word for Private repository

### ISSUE THREAD

**Issue by jules-ch** [python-poetry/poetry#4920]
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.

## Issue

There is no mention of the source keyword to specify a custom private repository for a specific package in the [Dependency Specification](https://python-poetry.org/docs/dependency-specification/) section.

It should be added to provide information on how to specify a specific source for a package.

**Comment by pierresouchay** [comment]
Yes, please!

I don't get how to query a private repository for only a few packages (and rely on PyPI by default for instance)

**Comment by jules-ch** [comment]
Yes, you need to define your source & add it in the your package definition.

```toml

[[tool.poetry.source]]
name = "gitlab"
url = "https://gitlab.com/api/v4/projects/<redacted>/packages/pypi/simple/"
secondary = true

[tool.poetry.dependencies]
python = ">=3.8,<3.9"
my_package_example = {version="0.1.0", source="gitlab"}

**Comment by pierresouchay** [comment]
yes, I finally found this, but I also found it does not work as I would expect.

I have the same exact example than you, but when I do a `poetry update`

=> all packages (not only `my_package_example` are looked in the `gitlab` repository while I would expect that secondary = true avoids this.
=> in the end, secondary does not work (at least it is performing requests against the repo despite its "secondary" attribute.

I would like basically to use one named "gitlab" to host 1 package and not trying to download every package on earth on it.

Does not seem to work for now (at least with poetry 1.1.9)
